swift/Samples/ModalContainer/Sources/ModalContainerViewController.swift

```diff
@@ -0,0 +1,333 @@
+
+        // Sort our views. We go front to back to allow dismissed views to appear above currently presented views (this will matter after transition support is added).
+        for modal in presentedScreens.reversed() {
+            view.sendSubviewToBack(modal.viewController.view)
+            if let dimmingView = modal.dimmingView {
+                view.sendSubviewToBack(dimmingView)
+            }
+        }
+
+        view.sendSubviewToBack(baseScreenViewController!.view)
+
+        setNeedsStatusBarAppearanceUpdate()
+
+        if #available(iOS 11.0, *) {
+            setNeedsUpdateOfHomeIndicatorAutoHidden()
+            setNeedsUpdateOfScreenEdgesDeferringSystemGestures()
+        }
+
+        // Set the topmost screen to be the accessibility modal
+        presentedScreens.last?.viewController.view.accessibilityViewIsModal = true
+        for modal in presentedScreens.dropLast() {
+            modal.viewController.view.accessibilityViewIsModal = false
+        }
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        baseScreenViewController?.view.frame = view.bounds
+
+        presentedScreens.forEach {
+            let displayInfo = ModalDisplayInfo(containerSize: view.bounds.size, style: $0.style)
+            $0.viewController.view.frame = displayInfo.frame
+            $0.dimmingView?.frame = view.bounds
+        }
+    }
+
+    override var childForStatusBarStyle: UIViewController? {
+        return topmostScreenViewController
+    }
+
+    override var childForStatusBarHidden: UIViewController? {
+        return topmostScreenViewController
+    }
+
+    override var childForHomeIndicatorAutoHidden: UIViewController? {
+        return topmostScreenViewController
+    }
+
+    override var childForScreenEdgesDeferringSystemGestures: UIViewController? {
+        return topmostScreenViewController
+    }
+
+    public override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+        return topmostScreenViewController?.supportedInterfaceOrientations ?? super.supportedInterfaceOrientations
+    }
+
+}
+
+fileprivate struct ModallyPresentedScreen {
+    var viewController: DescribedViewController
+    var screenType: Screen.Type
+    var style: ModalContainerScreen.Modal.Style
+    var key: String
+    var dimmingView: UIView?
+
+    var identifier: ModalIdentifier {
+        return ModalIdentifier(
+            screenType: screenType,
+            style: style,
+            key: key
+        )
+    }
+}
+
+extension ModalContainerScreen.Modal {
+    fileprivate var identifier: ModalIdentifier {
+        return ModalIdentifier(
+            screenType: type(of: screen),
+            style: style,
+            key: key
+        )
+    }
+}
+
+fileprivate struct ModalIdentifier: Hashable {
+    var screenType: Any.Type
+    var style: ModalContainerScreen.Modal.Style
+    var key: String
+
+    static func == (lhs: ModalIdentifier, rhs: ModalIdentifier) -> Bool {
```

Comment:
Can this be auto-synthesized?

Reply:
Not sure if auto synthesis is possible since some properties in the struct are not hashable.